index.js

```diff
@@ -0,0 +1,145 @@
+const execFile = require('child_process').execFile;
+
+const helmBinary = process.env.HELM_BIN;
+const args = process.argv;
+
+/**
+ * Prints the extension's usage
+ */
+function printUsage() {
+  const usage = "Converts helm's output to json format.\n"
+    + 'Works for commands: install, status\n'
+    + '\n'
+    + 'Example usage:\n'
+    + '  helm json install stable/rabbitmq\n'
+    + '  helm json status my-release-name\n';
+  process.stdout.write(usage);
+}
+
+// The expected arguments are:
+// ['node', 'js file', 'command', 'chart / release name']
+if (args.length < 4) {
+  printUsage();
+  process.exit(0);
+}
+
+/**
+ * Builds a json object from a string array of resources.
+ * format example:
+ * resourcesArr = ['==> v1beta1/StatefulSet \n
+                    NAME   DESIRED  CURRENT  AGE \n
+                    fun-elk-mariadb  1        1        2s',
+                   '==> v1/Pod(related) \n
+                    NAME   READY  STATUS   RESTARTS  AGE \n
+                    fun-elk-wordpress-665ff69d4b-9kg8x  0/1    Pending  0  2s \n
+                    fun-elk-mariadb-0    0/1    Pending  0   2s']
+ */
+function ConvertToObject(resourcesArr) {
+  const json = [];
+  resourcesArr.forEach((element) => {
+    const lines = element.split('\n');
+    let name;
+    const resources = [];
+    lines.forEach((line) => {
+      if (line.startsWith('==>')) {
+        name = line.substring(4).trim();
+      } else if (line.startsWith('NAME') === false) {
+        const trimmedLine = line.trim();
+        resources.push(trimmedLine.substring(0, trimmedLine.indexOf(' ')));
+      }
+    });
+
+    json.push({
+      name,
+      resources,
+    });
+  });
+  return json;
+}
+
+/**
+ * From the bulk resources string, parse each individual resource and insert
+ * it to the resources array which is returned at the end.
+ */
+function ParseResources(resourcesStr) {
+  const array = [];
+
+  // Sanity
+  if (resourcesStr.trim() === '') {
+    return array;
+  }
+
+  let ind = 0;
+  let outputCopy = resourcesStr;
```

Comment:
This has no meaning, since resourcesStr doesn't change. if you want to make a copy, use [slice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/slice)